### ISSUE TITLE
Cannot Build With Xcode 11.3.1 - No such module 'RescueBroadcast'

### ISSUE THREAD

**Issue by rsymons22** [LogmeinRescue/iOS-SDK#48]
- Upgraded to Xcode 11.3.1
- Seeing Swift Compiler Error "No such module 'RescueBroadcast'" in SampleHandler.swift
- Removed and reinstalled RescueSDK/Core and RescueSDK/Broadcast using CocoaPods. Now using version 5.5.
- Restarted Xcode, performed a clean of all derived data, still seeing the same compiler error.
- Code worked fine under Xcode 10.3 but Apple is requiring all apps to be submitted using Xcode 11.x starting next month.
- Note that we are NOT using LMICoreMedia.

**Comment by tibiur** [comment]
Do you have latest the CocoaPods version (1.9.1) ?

**Comment by rsymons22** [comment]
Thanks for the quick reply. I was using the CocoaPods Mac app which I do not believe always contains the latest version of Cocopods. After installing the v1.9.1 command line tools, I'm now seeing the following error when I try to build in Xcode (11.3.1):

Module compiled with Swift 5.0.1 cannot be imported by the Swift 5.1.3 compiler: /Applications/SSS_Apps/FreedomCarePlus/Pods/RescueSDK/Frameworks/RescueBroadcast.framework/Modules/RescueBroadcast.swiftmodule/arm64.swiftmodule

**Comment by tibiur** [comment]
Please make sure, you are using version 5.5 of our SDK. Release 5.5 has module compatibility and can be linked with projects using any swift 5 compiler.

It looks like you are using and older version. In your error message there is a path: /Applications/SSS_Apps/FreedomCarePlus/Pods/RescueSDK/Frameworks/RescueBroadcast.framework/Modules/RescueBroadcast.swiftmodule/arm64.swiftmodule

It shows that you are linking with an older version of the Rescue iOS SDK. v5.5 uses the new XCFramework format. Path should look like:
/Applications/SSS_Apps/FreedomCarePlus//Pods/RescueSDK/Frameworks/RescueBroadcast.xcframework/ios-arm64/RescueBroadcast.framework/Modules/RescueBroadcast.swiftmodule/arm64.swiftmodule

**Comment by tibiur** [comment]
Just double checked:
-downloaded the broadcast example
-removed the manually linked Rescue iOS frameworks
-added a podfile:

target 'RescueBroadcastExample' do
  use_frameworks!
  pod 'RescueSDK/Core'
  pod 'RescueSDK/Broadcast'
end

target 'RescueBroadcastExampleExtension' do
  use_frameworks!
  pod 'RescueSDK/Broadcast'
end

-did setup bundle id and signing
-used xcode 11.3.1 to build and run on iPhone X

The project works for me.

**Comment by rsymons22** [comment]
Thanks. Even though cocoapods was saying that I had v5.5 of the SDK, it clearly wasn't. After a pod deintegrate, pod clean and pod install, things seem to be working fine now. This is all for the Freedom Care apps by the way in the United States. I'm sure they'd like me to pass along their thanks for all your help. Stay healthy!!!

**Comment by tibiur** [comment]
I am glad I was able to help.

**Comment by StefanVanStaden** [comment]
@tibiur I managed to get our app working by manually embedding the frameworks but then it does not want to upload to the app store.

Followed all steps above to try and install via pods but still getting the module not found error message.

**Comment by tibiur** [comment]
Hi! Do you have some logs?

**Comment by StefanVanStaden** [comment]
@tibiur When installing from Cocoa Pods we only get the No such module 'RescueBroadcast' error message.

**Comment by tibiur** [comment]
You have to embed  the frameworks to your application, not to the extension. Your pod file should look like this:

target 'YourApplication' do
use_frameworks!
pod 'RescueSDK/Core'
pod 'RescueSDK/Broadcast'
end

target 'YourExtension' do
use_frameworks!
pod 'RescueSDK/Broadcast'
end

**Comment by StefanVanStaden** [comment]
We only embedded the Rescuebroadcast framework to the extension as this is the only way we could get rid of the No such module error. we did not use pods in the case of manually embedding the frameworks.

I just retried with the following podfile

Our Pod file does look like that see below. 
target 'Ignition' do
 
  use_frameworks!
  pod 'RescueSDK/Core'
  pod 'RescueSDK/Broadcast'


end

target 'IgnitionBroadcastExtension' do
  use_frameworks!
  pod 'RescueSDK/Broadcast'

end

Which outputs 
`Analyzing dependencies
Downloading dependencies
Installing RescueSDK 5.5
Generating Pods project
Integrating client project
Pod installation complete! There are 2 dependencies from the Podfile and 1 total pod installed.

[!] Automatically assigning platform `iOS` with version `12.0` on target `Ignition` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.

[!] Automatically assigning platform `iOS` with version `12.0` on target `IgnitionBroadcastExtension` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.`

We then still receive the no such module found error.

**Comment by tibiur** [comment]
We use the new XCFramework format, so there is no need for the strip-frameworks script. Please make sure you are using cocoapods version  1.9.1 at least

**Comment by StefanVanStaden** [comment]
@tibiur  I was indeed on an older version of cocoa pods (1.8.x). Updating did help sort out most issues we still received the invalid bundle issue. We fixed this by changing the following build settings.

Enable Bit code - NO (Main application and extension target)
Always Embed standard swift libraries - YES (Main application target)
Always Embed standard swift libraries - NO (Extension target)

We also had to remove all frameworks that were linked and embedded on the extension target.

This allowed us to archive and upload to iTunes connect store
